R/expect-no-condition.R

```diff
@@ -0,0 +1,113 @@
+#' Does code running without error, warning, message, or other condition?
+#'
+#' These expectations assert the opposite of [expect_error()],
+#' `expect_warning()`, `expect_message()`, and `expect_condition()`, namely
+#' that the code runs without error, warning, or message respectively.
+#'
+#' @inheritParams expect_error
+#' @param message,class The default usage with `message = NULL, clas = NULL`
+#'   will match any error/warning/message/condition. This is a broad assertion
+#'   so you'll typically want to make it narrower by either supplying `message`
+#'   (a regular expression that must match the message of the condition) or
+#'   `class` (which the condition must inherit from).
```

Comment:
```suggestion
#'   will match any error/warning/message/condition. This is a broad assertion,
#'   so sometimes you may want to be more specific about the condition you are
#'   hoping to **not** see, i.e. the condition that motivates the expectation.
#'   Similar to `expect_error()` and friends, you can specify the `message` (a
#'   regular expression that the message of the condition must match) and/or the
#'   `class` (a class the condition must inherit from).
```

My main comment in this area is that the significance of `message` and `class` are a bit mind-bend-y here and that requires careful wording.

In the positive case (so: `expect_error()`), you're saying what you *expect* to see.

In this negative case, you're describing the condition you're worried about, i.e. the one that motivated you to write this expectation.

I don't think the current wording is wrong *per se*, but I've proposed an alternative

I also think one is much less likely to specify `message` or `class` for the `expect_without_*()` expectation and that strikes me as totally OK.

Reply:
Moved to slack for quicker iteration.